### PR TITLE
Allow user to cache their credentials

### LINKED ISSUE #42: Users want to stay logged in instead of having their credentials expire
Some of our users don't like that they get logged out every 24 hours. The initial fix was to ask for less information when they got timed out, however, they still found the process of logging in frustrating. New idea is to allow the option to cache credentials so that users don't have to log in unless they chose to have their credentials expire every 24 hours or if they log out.

### PR DESCRIPTION
**Pull request by samlevy3**
We will not have a config option where users can select whether they want to cache their credentials or not. If they chose to cache them, then they will not get logged out every 24 hours. Closes #42

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -323,6 +332,18 @@ function getConfig(request) {
...
   
+  config.newInfo()
+  .setId('Caching Credentials')
+  .setText('You have set credentials to be cached: ' +  user.getProperty('dscc.areCredentialsCached'));
+  
+  config
+  .newSelectSingle()
+  .setId('cache_credentials')
+  .setName('Would you like to save your credentials?')
+  .setHelpText('Chose whether to cache your ODK username and password to avoid getting logged out every 24 hours. You can leave this blank if it is set to what you want it to be.')
```
Comment (lognaturel) [inline src/main.js:343]: How about adding a note to clarify that this means their Central password is stored on Google servers? Maybe something like "This means your Central password will be stored on Google servers. We recommend using a Project Viewer account."

> You can leave this blank if it is set to what you want it to be.
Not immediately understanding this! 'no' will be selected by default, right? So it won't be blank?

Comment (lognaturel) [inline src/main.js:343]: Chose -> Choose

File: src/main.js [hunk 2]
```diff
@@ -136,6 +136,9 @@ function setCredentials(request) {
 * @returns {boolean} whether the username + password + path are correct
 */
 function validateAndStoreCredentials(username, password, path) {
+  Logger.log(username);
```
Comment (lognaturel) [inline src/main.js:139]: Remove or wrap with debug?

File: src/main.js (hunk 1 above)
Comment (samlevy3) [reply src/main.js:343]: Oooh, I can definitely add that! And I need to play around with the language, but google data studio doesn't allow default options for the dropdown. I have a textbox above the select that lists their current selection (caching or no caching) and if they don't select anything then it leaves their current selection. That is why I said they could leave it blank.

File: src/main.js [hunk 3]
```diff
@@ -136,9 +136,6 @@ function setCredentials(request) {
 * @returns {boolean} whether the username + password + path are correct
 */
 function validateAndStoreCredentials(username, password, path) {
-  Logger.log(username);
```
Comment (samlevy3) [inline src/main.js]: Don't know how I missed this! Thanks for the catch.

File: src/main.js [hunk 4]
```diff
@@ -334,13 +331,16 @@ function getConfig(request) {
   
   config.newInfo()
   .setId('Caching Credentials')
-  .setText('You have set credentials to be cached: ' +  user.getProperty('dscc.areCredentialsCached'));
+  .setText('Currently, your credentials are [' + (user.getProperty('dscc.areCredentialsCached') == 'true' ? 'being cached' : 'not being cached') + ']. If you ' +
```
Comment (samlevy3) [inline src/main.js:334]: GDS unfortunately does not allow a lot of text configuration so I cannot bold statements. The idea here is to tell the user whether they are currently caching their password or not. This choice is the default if they do not select a new choice.

File: src/main.js (hunk 1 above)
Comment (lognaturel) [reply src/main.js]: I see! I had the wrong UI in my head -- I thought that was a dialog. Makes sense, then!

**Pull request merged**